File: UI/src/app/dtos/Master/employeeDto.ts
import { BaseEntityDto } from "../base/baseEntityDto";

export class EmployeeDto extends BaseEntityDto {
    firstName: string | null;
    lastName: string | null;
    fileName: string | null;
    departmentName: string | null;
    departmentId: number | null;
    designationName: string | null;
    designationId: number | null;
    contact: number | null;
    joinDate: string;
    basicSalary: number | null;
    tax: number | null;
    totalWorkingDays: number | null;
    leave: number | null;
    receivedSalary: number | null;
    salaryTotal: number | null;
    firstNameRequired: string | null;
    lastNameRequired: string | null;
    designationRequired: string | null;
    departmentNameRequired: string | null;
    contactRequired: string | null;
    salaryRequired: string | null;
}

File: API/Dtos/Masters/EmployeeDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using API.Dtos.Base;

namespace API.Dtos.Masters
{
    public class EmployeeDto  : BaseEntityDto
    {        
        public string? FirstName{get;set;}
        public string? LastName {get;set;}
        
        public string? fileName {get;set;}
        public string? DepartmentName {get;set;}
        public int? DepartmentId {get;set;}

        public string? DesignationName {get;set;}
        public int ? DesignationId {get;set;}
        
        public int ? Contact {get;set;}

        public DateTime JoinDate {get;set;}
        public int? BasicSalary {get;set;}

        
        public int ? Tax {get;set;}
        public  int ? totalWorkingDays{get;set;}
        public int? Leave{get;set;}
        public double? receivedSalary{get;set;}
        public double ?salaryTotal{get;set;}

        public string? FirstNameRequired{get;set;}
        public string? LastNameRequired{get;set;}
        public string? DesignationRequired{get;set;}
        public string? DepartmentNameRequired{get;set;}
        public string? ContactRequired{get;set;}

           public string? SalaryRequired{get;set;}
        
    }
}